department-of-veterans-affairs/vets-website
# Add copy of addDisabilities page behind feature flag

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

Summary
Add a copy of the addDisabilities page to start modifying
Place copy, addDisabilitiesRevised behind a feature flag
update depends for existing addDisabilities page, so it's not shown when the new page is shown
unit test helper functions for determining which page is shown


## Related issue(s)

- https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/436

## Testing done

- No old behavior for addDisabilities page should be changed when feature flag is OFF

- Cypress used for debugging failing `depends` logic when developing this feature.
- ensure the feature toggle `disability_526_improved_autosuggestions_add_disabilities_page` is **disabled** 
  - form should work as it does in production
  - Open the Cypress UI `yarn cy:open`
  - navigate to `toxic-exposure.cypress.spec.js` and `all-claims.cypress.spec.js`
  - run both (double click each test)
  
- title indicates revised version of the page is in-use when feature flag is ON
- ensure feature toggle is **enabled**
- navigate to to VA Benefits and Health Care -> Disability -> File a claim online -> File a disability claim online
- when going through the 526-ez, make sure you select the option to file for NEW conditions.
- Observe the title displays "REVISED VERSION" (screenshot below)


## Screenshots
![Screenshot 2024-05-07 at 9 28 45 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/17188013/4c8d15fd-400c-4d11-b326-f94847af30a9)
![Screenshot 2024-05-07 at 9 29 19 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/17188013/ab13e984-6a76-498c-8d2a-54dfca99160d)


_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
